### ISSUE TITLE
Improve QSVR unit tests 

### ISSUE THREAD

**Issue by adekusar-drl** [qiskit-community/qiskit-machine-learning#434]
### What is the expected enhancement?

Currently unit test in [TestQSVR](https://github.com/Qiskit/qiskit-machine-learning/blob/main/test/algorithms/regressors/test_qsvr.py) are based on the classification dataset, e.g., labels are `[0, 0, 1, 1]`. Such a dataset is not very well suited to test regressors. This issue intends to fix this problem and implement unit tests on a more suitable dataset.

**Comment by adekusar-drl** [comment]
@AbdullahKazi500 do you have ideas what exactly to be fixed in this issue?

**Comment by adekusar-drl** [comment]
Sure, I can. But the question I posted above is still valid.

**Comment by adekusar-drl** [comment]
Thanks for getting back. A few comments:
- A dataset should be really small, like a couple of data points. Currently, there are 4 of them in the test. One of the ideas to try out is to use the regression dataset from https://github.com/Qiskit/qiskit-machine-learning/blob/main/docs/tutorials/02_neural_network_classifier_and_regressor.ipynb. But you are welcome to propose something else.
- I suggest to modify existing test case unless it is inconvenient to update them.
- There's no need to split the dataset, these are just tests.
- One metric is enough. Currently, tests compare evaluated score vs true value.
- Not sure hyperparameterization is required here.

So, the work should be pretty straight forward. Let me know if you have questions.

**Comment by adekusar-drl** [comment]
No, all of them should fixed to certain values and that's it. Tests must check that algorithms work and results are meaningful, but no more.

**Comment by adekusar-drl** [comment]
@AbdullahKazi500 is there any progress here?

**Comment by edoaltamura** [comment]
Hi @AbdullahKazi500 has there been any recent progress? If so, please organize and submit the commits in a PR.

**Comment by edoaltamura** [comment]
I haven't heard recent updates on this test update and wanted to touch base and share a few timeline considerations FYI. We aim to include this update in the v.0.8 initial release, which is expected around the end of October 2024. This means that we would like to support long-standing community contributions - like this one - pull-requested by Friday 25 October, and leave enough time to coordinate the merging and release preparation.

Are you still available to work on this with the suggestions in https://github.com/qiskit-community/qiskit-machine-learning/issues/434#issuecomment-1476375330 by then or rather step out on this occasion?

**Comment by RishiNandha** [comment]
Greetings, I'm working on this issue at the moment. I read the discussion above. I just wanted to confirm that I am not oversimplifying the task. From what I understand:

1. We need to update the self.sample_train, self.label_train... to a small (about 10 samples) regression type datapoints rather than classification (there is a reference notebook attached in the discussion above)

2. Find the expected score we should be getting from these, and verify that all the existing tests defined work appropriately

That works right?

**Comment by edoaltamura** [comment]
Hi @RishiNandha, yes, you're on the right track with your approach. To summarise:

- Update training data: replace the current classification dataset (`[0, 0, 1, 1]`) with a small regression dataset, ideally fewer than 10 samples. this ensures the data aligns with regression testing needs. ​

- Determine expected scores: establish expected performance metrics for the new regression dataset to serve as benchmarks for accuracy and reliability.​

- Verify existing tests: ensure all current tests function correctly with the updated regression data, confirming that the modifications haven't introduced any issues.

@AbdullahKazi500 Thanks for your keen interest. At this time, we have internally proposed Rishi to take the lead on this issue. Feel free to comment or add your contribution on top of theirs, we'd love to see this kind of collaboration.

**Comment by rockywick** [comment]
Hi @edoaltamura, 

I am working with @RishiNandha on this issue. I noticed that the test_qsvr.py and the test_fidelity_quantum_kernel_qsvr.py files seem to be almost identical except for very minor changes (assert precision and explicitly enforcing the kernel matrix to be positive semi-definite in test_qsvr.py). I just wanted to clarify if I am missing something and wanted to understand what the purpose of the test_fidelity_quantum_kernel_qsvr.py file is.

Thank you

**Comment by edoaltamura** [comment]
Hi @rockywick, thanks for the PR! Fair point, the tests are almost identical indeed. @OkuyanBoga @oscar-wallis do you know if this was the case before the primitive V2 upgrade?

**Comment by edoaltamura** [comment]
@rockywick could you please open a new issue reporting what you flagged (below)?

> I noticed that the test_qsvr.py and the test_fidelity_quantum_kernel_qsvr.py files seem to be almost identical except for very minor changes (assert precision and explicitly enforcing the kernel matrix to be positive semi-definite in test_qsvr.py). I just wanted to clarify if I am missing something and wanted to understand what the purpose of the test_fidelity_quantum_kernel_qsvr.py file is.

This way, we'll remind ourselves to come back to discussing this particular point, and merge your #900 in the meantime!
